basket/news/tasks.py

```diff
@@ -1335,13 +1336,14 @@ def upsert_amo_user_data(data):
     # include the ID in update or add since we couldn't find
     # the user with this ID above
     amo_data['amo_id'] = amo_id
-    user = get_user_data(email=email, extra_fields=['id'])
+    user = get_user_data(fxa_id=fxa_id, extra_fields=['id'])
     if user:
         sfdc.update(user, amo_data)
         # need amo_id for linking addons and authors
         user['amo_id'] = amo_id
         return user
 
+    # User not found in database
```

Comment:
This the section below for the case user is not found in the database with either `amo_id` or `fxa_id`? 

Could this ever be true, given that AMO users sign in with FxA now? If yes maybe we should request in https://github.com/mozilla/addons-server/issues/14318 to send both FxA ID *and* email so we can create the record @pmac?

Reply:
It shouldn't ever be true, but it's possible some user wasn't synced properly with SFDC so I think we have to deal with the possibility. We could just ignore these, but that also seems suboptimal. Maybe we should ask for email. I think I remember us discussing this and deciding against it, but I don't recall why. @diox do you remember?

Reply:
Well the idea was that FxA is always the source of truth - we shouldn't need to store the email elsewhere. It seems even more suboptimal to have another field...

Reply:
It wouldn't mean another field. What we're wondering is if we should create a record in the CRM in the case that we find no record for the `fxa_id` we receive for you. It's possibly that the sync from FxA wasn't perfect. This should almost never happen. But if it did and you also sent us email then we could create a record for the user. If we have no email and we can't find the FxA ID then we have to just ignore the update.

Reply:
Yeah, still not sure I like that very much. It would hide the problem, but I'm concerned about the possibility of opening up new edge cases...

Reply:
Updated code to ignore this situation, since we don't have the email address. We can increase a statsd counter and/or log an error to Sentry if we see value.